bin/simple_engine.ml

```diff
@@ -28,7 +33,10 @@ let runtime_begin device tones _domain_id ts event =
   | Some ts ->
       Midi.(
         write_output device
-          [ message_on ~note ~timestamp:ts (* ~volume:'\070' *) () ]);
+          [
+            message_on ~note:note.Scale.note ~timestamp:ts
```

Comment:
To make this less heavy syntactic wise, you may directly deconstruct the noteAndVolume record when you are binding it with `let`.

Instead of doing `let note  =`, you can do `let {note; volume} = Play.event_to_note` and have the right variable names directly available to you. (if you rename `vol` in your record to `volume`, that is.)

Reply:
@Engil , I don't understand how to implement this

```
let runtime_begin device tones _domain_id ts event =
  let {note, volume} = Play.event_to_note tones event in   <---- unbound record field note
  match adjust_time ts with
  | None -> ()
  | Some ts ->
        write_output device
          [
            message_on ~note:note ~timestamp:ts
              ~volume:volume ();
          ]);
      Printf.printf "%f: start of %s. ts: %ld\n%!" (Sys.time ())
        (Runtime_events.runtime_phase_name event)
        ts
```

I also tried 
```
let runtime_begin device tones _domain_id ts event =
  Midi.(
  let Scale.note_data{note, volume} = Play.event_to_note tones event in
  match adjust_time ts with
  | None -> ()
  | Some ts ->
        write_output device
          [
            message_on ~note:note ~timestamp:ts
              ~volume:volume ();
          ]);
      Printf.printf "%f: start of %s. ts: %ld\n%!" (Sys.time ())
        (Runtime_events.runtime_phase_name event)
        ts
```
which also didn't work.
And using directly as list, doesn't work as the type changes from `note_data` to `char * char`
```
let runtime_end device tones _domain_id ts event =
  let note, volume = Play.event_to_note tones event in
  match adjust_time ts with
  | None -> ()
  | Some ts ->
      Midi.(
        write_output device
          [
            message_off ~note:note ~timestamp:ts
              ~volume:volume ();
          ]);
      Printf.printf "%f: start of %s. ts: %ld\n%!" (Sys.time ())
        (Runtime_events.runtime_phase_name event)
        ts
```
```

Reply:
Can you help with resolving this!

Reply:
Shouldn't it be `let { note; vol }` ? (Notice the ;) :-)

Reply:
I completely missed this hehe. New to Ocaml :3

But I'm still getting the same error
```
 dune build                                                                                                                                                                                     ─╯
File "bin/simple_engine.ml", line 30, characters 7-11:
30 |   let {note; volume} = Play.event_to_note tones event in
            ^^^^
Error: Unbound record field note
```